### PR TITLE
Projects controller vs repositories

### PR DESCRIPTION
**Pull request by vladak**
Fixes a problem when setting per project property that is not shadowed by repository property.

### CODE REVIEW

File: opengrok-web/src/main/java/org/opengrok/web/api/v1/controller/ProjectsController.java [hunk 1]
```diff
@@ -311,13 +311,20 @@ public void set(
...
+            // Refresh field values for project's repositories for this project as well.
             List<RepositoryInfo> riList = env.getProjectRepositoriesMap().get(project);
             if (riList != null) {
                 for (RepositoryInfo ri : riList) {
                     Repository repo = getRepository(ri, CommandTimeoutType.RESTFUL);
 
-                    // set the property
+                    // Set the property if there is one.
+                    try {
+                        ClassUtil.getFieldValue(repo, field);
```
Comment (ahornace) [inline opengrok-web/src/main/java/org/opengrok/web/api/v1/controller/ProjectsController.java:322]: Shouldn't the `setFieldValue` be called here and the invocation below removed?

Comment (ahornace) [reply opengrok-web/src/main/java/org/opengrok/web/api/v1/controller/ProjectsController.java:322]: Or, if we want to propagate errors, then maybe introduce a new method `hasField` or similar?

Comment (vladak) [reply opengrok-web/src/main/java/org/opengrok/web/api/v1/controller/ProjectsController.java]: Definitely want to propagate the errors. While adding `hasField()` I realized the controller actually sets the values on wrong repository object set. Setting the values on `Repository` objects constructed from `RepositoryFactory` will not have any effect on the configuration. The test was wrong also.

**Comment by ahornace** [comment]
Looking good.

**Pull request merged**